operations/iuf/stages/update_vcs_config.md

```diff
@@ -0,0 +1,144 @@
+default:
+  working_branch: "integration-{{version_x_y_z}}"
+```
+
+Each product that utilizes a working branch will contain a `working_branch` entry in `product_vars.yaml`, for example:
+
+```yaml
+cos:
+  version: 2.4.86
+  working_branch: "{{ working_branch }}"
+```
+
+In this example, the COS reference to `{{ working_branch}}` will be substituted with the value specified in the default section for `working_branch`. Thus, the COS value for `working_branch` will be `integration-{{version_x_y_z}}`,
+which will in turn be substituted with the actual product value, resulting in the final value `integration-2.4.86`.
+
+### Site Variables
+
+Site variables, typically specified in a `site_vars.yaml` file, allow the administrator to override values provided by recipe variables, including both default values and product-specific entries.
+
+Variable substitutions can be used in `site_vars.yaml` to specify the desired branching values, minimizing the need to make updates to the file. For example, to change the working branch for all products that use a default
+working branch, the administrator can simply provide a default section in `site_vars.yaml` with a new value for `working_branch`. In the following example, adding this section in `site_vars.yaml` will result in all products
+using the default branch to only use the `x.y` portion of the version instead of the `x.y.z` values specified in the `product_vars.yaml` file.
+
+```yaml
+default:
+  working_branch: "integration-{{version_x_y}}"
+```
+
+In a similar fashion, product-specific entries can be made in `site_vars.yaml` to override individual products, for example:
+
+```yaml
+cos:
+  working_branch: "test-{{ version_x_y_z }}"
+```
+
+Remember that content in `site_vars.yaml` overrides entries in `product_vars.yaml`, e.g. if a product entry in `site_vars.yaml` contained a `version` entry, that value would mask the `version` entry present in `product_vars.yaml`.
+
+### Session Variables
+
+Session variables are the set of product/version combinations being installed by the current activity. These values are internal to `iuf`.
+
+### Mechanics
+
+`iuf` builds a set of effective site variables utilized by back-end IUF operations by performing a set of merges and substitutions:
+
+1. The base set of data is created from the most recent product entries in the product catalog.
+1. Recipe variables, if specified, will be merged, overriding any matching entries. **Caveat**: Any product specified with the `iuf run` `--mask-recipe-prods` argument will be omitted from this merge, i.e. the version found in step 1 will be used instead.
+1. Site variables, if specified, will be merged, overriding any matching entries.
+1. Session variables will be merged, overriding any matching entries.
+
+Once `iuf` has performed the merges, it performs substitutions based on the following supported variables:
+
+```yaml
+{{name}} - product name from the product manifest
+{{version}} - version from the product manifest
```

Comment:
Since the use of the full `{{version}}` string makes branch matching in the `update-vcs-config` template, please remove this line: `{{version}} - version from the product manifest`

Reply:
fixed